sw/device/lib/testing/test_framework/FreeRTOSConfig.h

```diff
@@ -25,12 +25,9 @@
 
 // Memory
 #define configCHECK_FOR_STACK_OVERFLOW 0
-#define configMINIMAL_STACK_SIZE 512  // in words
+#define configMINIMAL_STACK_SIZE 256  // in words
+#define configTOTAL_HEAP_SIZE ((size_t)0x1500u)
```

Comment:
Per our offline discussion, please investigate the malloc failure hook so we can print a useful message in the event that a test exceeds the configured heap size.

Reply:
Didn't realize I already had the malloc failure hook (mostly) implemented, just disabled (silly me!)

I have added an additional commit to enable this malloc failure hook (and print a more useful message). Additionally I have added a stack overflow hook that prints a useful message as well.

Lmk what you think!

Marking this as fixed.